providers/datasource.rb

```diff
@@ -55,12 +55,16 @@ def do_create(exists, allow_update, should_update, grafana_options)
     end
   else
     converge_by("Creating data source #{new_resource.name}") do
-      add_data_source(new_resource.source, grafana_options)
+      add_data_source(new_resource.source, legacy_http_semantic, grafana_options)
     end
     Chef::Log.info "Added #{new_resource.source_name} as a datasource to Grafana"
   end
 end
 
+def legacy_http_semantic
+  node[:grafana][:version] < '2.0.3'
```

Comment:
It would be better to use the `Chef::Version` class for this, because comparing string would not do what's expected for version number:

```ruby
"1.0.2" > "1.0.10" # true, because 1 comes before 2 and we're using simple string comparison
Chef::Version.new("1.0.2") > Chef::Version.new("1.0.10") # false, because it compares each part of the version
```
Otherwise the change is good enough for me.